adi/adar1000.py

```diff
@@ -41,31 +43,374 @@ class adar1000(attribute, context_manager):
     parameters:
         uri: type=string
             URI of IIO context with ADAR1000(s)
+        context: type=iio.Context
+            IIO Context for the device. Only required if using adar1000_array.
         chip_id: type=string
             String identifying desired chip select and hardware ID for the
             ADAR1000. This string is the label coinciding with each chip
             select and hardware address and are typically in the form csbX_chipX.
             The csb line can be any number depending on how many are used in the
             system. The chip number will typically be 1-4 because each CSB line
             can control up to four ADAR1000s.
-        context_handle: type=iio.Context
-            IIO context handle. Not used unless creating an array of chips.
+        device_number: type=int
+            Integer indicating the device number in the array
+        array_element_map: type=list[list[int]]
+            List with the map of where the ADAR1000 channels are in the array.
+            Each entry in the map represents a row of array channels referenced
+            by element number. For example, a map:
+                | [[1, 5, 9, 13],
+                | [2, 6, 10, 14],
+                | [3, 7, 11, 15],
+                | [4, 8, 12, 16]]
+            represents an array of 16 elements (4 ADAR1000s) in a square array.
+        channel_element_map: type=list[int]
+            List of integers relating the array element numbers to the channels
+            of the ADAR1000. Each number in the list is the element number in the
+            larger array, in order of the ADAR1000 channels. For example, a list
+            [10, 14, 13, 9] indicates that the ADAR1000's channels are the
+            following elements in the full array:
+                | Channel 1: Element # 10
+                | Channel 2: Element # 14
+                | Channel 3: Element # 13
+                | Channel 4: Element # 9
     """
 
     _device_name = ""
+    _BIAS_CODE_TO_VOLTAGE_SCALE = -0.018824
+
+    class adar1000_channel:
+        """Class for each channel of the ADAR1000. This class is not meant
```

Comment:
Due to the instantiation comment can you add a check in __init__ like: 
```
if type(self) == adar1000_channel:
    raise something
```

Reply:
I can add this, but what do you think about just hiding it with a leading "_"?

Reply:
Added a _ prefix is fine.

Reply:
I should undo that then. It needs to show up in the doc so that people can see what properties/methods are available.

Reply:
I'm not sure I can add your check that you originally suggested, as I believe that will always fail? During __init__, won't self always equal adar1000_channel?